Dialog/index.d.ts

```diff
@@ -0,0 +1,46 @@
+import MaterialComponent from '../MaterialComponent';
+import { VNode } from 'preact';
+import Button from '../Button';
+import { MDCFoundation, MDCComponent } from '../MaterialComponentsWeb';
+
+declare interface IDialogProps {
+  onAccept?: () => void;
+  onCancel?: () => void;
+}
+export default class Dialog extends MaterialComponent<IDialogProps & JSX.HTMLAttributes, {}> {
+  static Header: typeof Header;
+  static Body: typeof Body;
+  static Footer: typeof Footer;
+  static FooterButton: typeof FooterButton;
+
+  MDComponent: MDCDialog;
+}
+
+declare class Header extends MaterialComponent<JSX.HTMLAttributes, {}> {}
+
+declare interface IBodyProps {
+  scrollable?: boolean;
+}
+declare class Body extends MaterialComponent<IBodyProps & JSX.HTMLAttributes, {}> {}
+
+declare class Footer extends MaterialComponent<JSX.HTMLAttributes, {}> {}
+
+declare interface IFooterButtonProps {
+  accept?: boolean;
```

Comment:
This particular prop seems problematic. It conflicts with `JSX.HTMLAttributes` where it is defined as string. When using it as boolean TypeScript complains `Type '{ accept: true; }' is not assignable to type 'HTMLAttributes'. Type 'true' is not assignable to type 'string'.`
And when I changed it to string, TS complains `Type '{ accept: "foo"; }' is not assignable to type 'IFooterButtonProps'. Type '"foo"' is not assignable to type 'boolean'.`

I'm not exactly sure how to fix it. I was wondering why you changed all those interfaces extending JSX.HTMLAttributes to just `<Props & JSX.HTMLAttributes, {}>` in the generic, did you run into similar issues with it or something?

Reply:
I experience the issue multiple places so I figured I would just do the same thing overall to make them as identical as possible.

Reply:
Ok, need to look into this a bit more and try to figure a way to override the type from JSX.HTMLAttributes.

Reply:
As the issue with `JSX.HTMLAttributes` we might need to create an almost identical interface but just without the `accept` property.